src/org/exist/storage/NativeBroker.java

```diff
@@ -1955,6 +1958,10 @@ public DocumentImpl storeTempResource(final org.exist.dom.memtree.DocumentImpl d
             } catch (final Exception e) {
                 LOG.warn("Failed to store temporary fragment: " + e.getMessage(), e);
                 //abort the transaction
+            } finally {
+                if(openedWithLock && temp != null) {
```

Comment:
please could you explain? `getOrCreateTempCollection()` does not create a lock on the new collection?